File: backend/src/main/java/com/hrms/employee/service/DesignationService.java
package com.hrms.employee.service;

import com.hrms.employee.dto.DesignationRequest;
import com.hrms.employee.dto.DesignationResponse;

import java.util.List;
import java.util.UUID;

public interface DesignationService {
    DesignationResponse create(DesignationRequest request);
    DesignationResponse getById(UUID id);
    List<DesignationResponse> getAll(UUID departmentId);
    DesignationResponse update(UUID id, DesignationRequest request);
    void delete(UUID id);
}

File: frontend/src/services/designationService.ts
import api from "./api"

export interface DesignationRequest { name: string; departmentId: string; description?: string }
export interface DesignationResponse { id: string; name: string; departmentId: string; departmentName: string; description?: string; createdAt: string }

const designationService = {
  getAll(departmentId?: string) {
    return api.get<DesignationResponse[]>("/designations", { params: departmentId ? { departmentId } : undefined })
  },
  getById(id: string) { return api.get<DesignationResponse>(`/designations/${id}`) },
  create(data: DesignationRequest) { return api.post<DesignationResponse>("/designations", data) },
  update(id: string, data: DesignationRequest) { return api.put<DesignationResponse>(`/designations/${id}`, data) },
  delete(id: string) { return api.delete(`/designations/${id}`) },
}

export default designationService
